programs/bpf_loader/src/syscalls.rs

```diff
@@ -2540,6 +2574,94 @@ fn call<'a>(
     Ok(SUCCESS)
 }
 
+// Return data hanlding
+pub struct SyscallSetReturnData<'a> {
+    invoke_context: Rc<RefCell<&'a mut dyn InvokeContext>>,
+    loader_id: &'a Pubkey,
+}
+impl<'a> SyscallObject<BpfError> for SyscallSetReturnData<'a> {
+    fn call(
+        &mut self,
+        addr: u64,
+        len: u64,
+        _arg3: u64,
+        _arg4: u64,
+        _arg5: u64,
+        memory_mapping: &MemoryMapping,
+        result: &mut Result<u64, EbpfError<BpfError>>,
+    ) {
+        if len > MAX_RETURN_DATA {
+            *result = Err(SyscallError::ReturnDataTooLarge(len, MAX_RETURN_DATA).into());
+            return;
+        }
+
+        let mut invoke_context = question_mark!(
+            self.invoke_context
+                .try_borrow_mut()
+                .map_err(|_| SyscallError::InvokeContextBorrowFailed),
+            result
+        );
+
+        let cost = invoke_context.get_compute_budget().cpi_bytes_per_unit;
+        question_mark!(
+            invoke_context.get_compute_meter().consume(len / cost),
+            result
+        );
+
+        let return_data = question_mark!(
+            translate_slice::<u8>(memory_mapping, addr, len, self.loader_id, true,),
+            result
+        );
+
+        invoke_context.set_return_data(return_data);
+
+        *result = Ok(0);
+    }
+}
+
+pub struct SyscallGetReturnData<'a> {
+    invoke_context: Rc<RefCell<&'a mut dyn InvokeContext>>,
+    loader_id: &'a Pubkey,
+}
+impl<'a> SyscallObject<BpfError> for SyscallGetReturnData<'a> {
+    fn call(
+        &mut self,
+        addr: u64,
+        len: u64,
+        _arg3: u64,
+        _arg4: u64,
+        _arg5: u64,
+        memory_mapping: &MemoryMapping,
+        result: &mut Result<u64, EbpfError<BpfError>>,
+    ) {
+        let invoke_context = question_mark!(
+            self.invoke_context
+                .try_borrow()
+                .map_err(|_| SyscallError::InvokeContextBorrowFailed),
+            result
+        );
+
+        let return_data = invoke_context.get_return_data();
+
+        let length = std::cmp::min(return_data.len() as u64, len);
```

Comment:
nit: move `std::cmp::min` to a `use` statement at the top

Reply:
Done